### PR TITLE
feat(v2/storage-2): implement decryption web worker and related hook

### PR DESCRIPTION
**Pull request by karrui**
This PR builds on work done in #3949

## Problem

This PR adds web workers handling in the React app, and implements the decryption web worker and its related hook. 

Related to #3129

## Solution

**Breaking Changes** 

- [x] No - this PR is backwards compatible  

**Features**:

- feat: implement decryption web worker
- feat: add StorageResponsesService
- feat: add new useDecryptionWorkers hook

**Improvements**:

- feat: use worker-loader in react by updating craco.config.js

## Before & After Screenshots
Pure business logic, no stories

## Tests

Not in this PR, will move AngularJS tests over in future PRs

### CODE REVIEW

**Review by siddarth2824 — APPROVED** [review]
LGTM

**Review by mantariksh — APPROVED** [review]
no departure from existing behaviour as far as I can tell!

File: frontend/src/features/admin-form/responses/ResponsesPage/storage/worker/decryption.worker.ts [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        csvRecord.setStatus(CsvRecordStatus.Unverified, 'Unverified')
+      }
+
+      if (downloadAttachments) {
+        let questionCount = 0
+
+        decryptedSubmission.forEach((field) => {
+          // Populate question number
+          if (field.fieldType !== 'section') {
+            ++questionCount
```
[the remark this answers is not in the document]
Comment (karrui) [reply frontend/src/features/admin-form/responses/ResponsesPage/storage/worker/decryption.worker.ts:153]: ya it was unused

File: frontend/src/features/admin-form/responses/ResponsesPage/storage/StorageResponsesService.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    delete params.endDate
+  }
+
+  const uriEncodedParams = mapValues(params, encodeURIComponent)
+
+  if (params) {
+    resUrl += Object.keys(uriEncodedParams)
+      .map((key) => `${key}=${uriEncodedParams[key]}`)
+      .join('&')
+  }
```
[the remark this answers is not in the document]
Comment (karrui) [reply frontend/src/features/admin-form/responses/ResponsesPage/storage/StorageResponsesService.ts]: thanks, greatly simplified. tested locally in 0bc531c

File: frontend/src/features/admin-form/responses/ResponsesPage/storage/worker/decryption.worker.ts [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+  const { line, secretKey, downloadAttachments } = data
+
+  let submission: StorageModeSubmissionStreamDto
+  let csvRecord: CsvRecord
+  const attachmentDownloadUrls: AttachmentsDownloadMap = new Map()
+  let downloadBlob: Blob
+
+  try {
+    submission = JSON.parse(line)
```
[the remark this answers is not in the document]
Comment (karrui) [reply frontend/src/features/admin-form/responses/ResponsesPage/storage/worker/decryption.worker.ts]: look at what you made me do 2cd5002

**Pull request merged**